homeassistant/components/binary_sensor/netatmo.py

```diff
@@ -0,0 +1,158 @@
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=10)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup access to Netatmo binary sensor."""
+    netatmo = get_component('netatmo')
+    home = config.get(CONF_HOME, None)
+
+    import lnetatmo
+    try:
+        data = WelcomeData(netatmo.NETATMO_AUTH, home)
+    except lnetatmo.NoDevice:
+        return None
+
+    if data.get_camera_names() == []:
+        return None
+
+    sensors = config.get(CONF_MONITORED_CONDITIONS, SENSOR_TYPES)
+
+    for camera_name in data.get_camera_names():
+        if config[CONF_CAMERAS] != []:
+            if camera_name not in config[CONF_CAMERAS]:
+                continue
+        for variable in sensors:
+            add_devices([WelcomeBinarySensor(data, camera_name, home,
+                                             variable)])
+
+
+class WelcomeBinarySensor(BinarySensorDevice):
+    """Represent a single binary sensor in a Netatmo Welcome device."""
+
+    def __init__(self, data, camera_name, home, sensor):
+        """Setup for access to the Netatmo camera events."""
+        self._data = data
+        self._camera_name = camera_name
+        self._home = home
+        if home:
+            self._name = home + ' / ' + camera_name
+        else:
+            self._name = camera_name
+        self._sensor_name = sensor
+        self._name += ' ' + sensor
+        self._unique_id = "Welcome_binary_sensor {}".format(self._name)
+        self.update()
+
+    @property
+    def name(self):
+        """The name of the Netatmo device and this sensor."""
+        return self._name
+
+    @property
+    def unique_id(self):
+        """Return the unique ID for this sensor."""
+        return self._unique_id
+
+    @property
+    def sensor_class(self):
+        """Return the class of this sensor, from SENSOR_CLASSES."""
+        return SENSOR_TYPES.get(self._sensor_name)
+
+    @property
+    def is_on(self):
+        """Return true if binary sensor is on."""
+        return self._state
+
+    def update(self):
+        """Request an update from the Netatmo API."""
+        self._data.update()
+        self._data.welcomedata.updateEvent(home=self._data.home)
+
+        if self._sensor_name == "Someone known":
+            self._state =\
+                    self._data.welcomedata.someoneKnownSeen(self._home,
+                                                            self._camera_name)
+        elif self._sensor_name == "Someone unknown":
+            self._state =\
+                self._data.welcomedata.someoneUnknownSeen(self._home,
+                                                          self._camera_name)
+        elif self._sensor_name == "Motion":
+            self._state =\
+                self._data.welcomedata.motionDetected(self._home,
+                                                      self._camera_name)
+        else:
+            return None
+
+
+class WelcomeData(object):
```

Comment:
This is a direct copy of the one in `camera` ? Why not make it a generic object, instantiated once in the main component?